open-webui: 0.6.18 -> 0.6.22

codgician opened the pull request:
Diff: https://github.com/open-webui/open-webui/compare/refs/tags/v0.6.18...refs/tags/v0.6.22

Changelog: https://github.com/open-webui/open-webui/blob/v0.6.22/CHANGELOG.md

Also fixed the update script so it could get the correct npmDepsHash.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

74k1 commented:
Hi @codgician

Did you test if the issue from https://github.com/NixOS/nixpkgs/issues/430433 is happening here as well?

codgician commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 433087`
Commit: `b7588ce8c2fb7501d1097ebc9c81726b0ff62251`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
  </ul>
</details>

codgician commented on pkgs/by-name/op/open-webui/update.sh:
```diff
@@ -14,9 +14,23 @@ tmpdir=$(mktemp -d)
+  update-source-version open-webui.frontend "${pyodide_version}" --file="$path" --version-key=pyodideVersion --source-key=pyodide
+fi
+
 rm -rf $tmpdir
```
Fixed in latest commit

codgician commented on pkgs/by-name/op/open-webui/update.sh:
```diff
@@ -14,9 +14,23 @@ tmpdir=$(mktemp -d)
 npm_hash=$(prefetch-npm-deps package-lock.json)
 sed -i 's#npmDepsHash = "[^"]*"#npmDepsHash = "'"$npm_hash"'"#' "$path"
+npm config delete legacy-peer-deps
+npm config delete force
```
Looked further into it and confirmed setting these flags does not affect the result of npmDepsHash. Removed them. Thanks!

codgician committed:
open-webui: 0.6.18 -> 0.6.22

Diff: https://github.com/open-webui/open-webui/compare/refs/tags/v0.6.18...refs/tags/v0.6.22

Changelog: https://github.com/open-webui/open-webui/blob/v0.6.22/CHANGELOG.md

74k1 commented:
Thanks @priegger 

--

I have just tested this Package (from `0.6.18` to `0.6.22`) along _with_ the changes from https://github.com/NixOS/nixpkgs/pull/431395 and https://github.com/NixOS/nixpkgs/pull/432897 already configured and I did not suffer any data loss.

This update looks good to me. It'd be helpful if anyone could also take a look at my PR for any improvements https://github.com/NixOS/nixpkgs/pull/432897.
